membership_prorrate/tests/test_membership_prorrate.py

```diff
@@ -13,36 +14,41 @@ def setUp(self):
                 'name': 'Membership product with prorrate',
                 'membership': True,
                 'membership_prorrate': True,
-                'membership_date_from': '2015-01-01',
-                'membership_date_to': '2015-12-31',
+                'membership_date_from': '2017-01-01',
+                'membership_date_to': '2017-12-31',
             })
         self.partner = self.env['res.partner'].create({'name': 'Test'})
 
     def test_create_invoice_membership_product_wo_prorrate(self):
         self.product.membership_prorrate = False
-        invoice = self.env['account.invoice'].create(
-            {'partner_id': self.partner.id,
-             'date_invoice': '2015-07-01',
-             'account_id': self.partner.property_account_receivable.id,
-             'invoice_line': [(0, 0, {'product_id': self.product.id,
-                                      'name': 'Membership w/o prorrate'})]}
-        )
-        self.assertEqual(invoice.invoice_line[0].quantity, 1.0)
+        invoice_id = self.partner.create_membership_invoice(
+            product_id=self.product.id, datas={'amount': 1.0})
+        invoice = self.env['account.invoice'].browse(invoice_id)
+        self.assertEqual(invoice.invoice_line_ids[0].quantity, 1.0)
 
     def test_create_invoice_membership_product_prorrate(self):
-        invoice = self.env['account.invoice'].create(
-            {'partner_id': self.partner.id,
-             'date_invoice': '2015-07-01',
-             'account_id': self.partner.property_account_receivable.id,
-             'invoice_line': [(0, 0, {'product_id': self.product.id,
-                                      'price_unit': 100.0,
-                                      'name': 'Membership with prorrate'})]}
-        )
+        invoice_line_account = self.env[
+            'account.account'
+        ].search([('user_type_id', '=', self.env.ref(
+            'account.data_account_type_expenses').id)], limit=1).id
```

Comment:
Well, it can serve, as it's an account and we don't need accounting correctness.